pkg/controller/clone/planner.go

```diff
@@ -532,3 +736,50 @@ func createDesiredClaim(namespace string, targetClaim *corev1.PersistentVolumeCl
 
 	return desiredClaim
 }
+
+func createTempSourceClaim(namespace string, targetClaim *corev1.PersistentVolumeClaim, snapshot *snapshotv1.VolumeSnapshot, client client.Client, ctx context.Context) (*corev1.PersistentVolumeClaim, error) {
+	// Attempting to get a storageClass compatible with the source snapshot
+	// but different from the one used in the target Claim.
+	// If unable to do so, we'll use the target Claim one anyway.
+	scName := *targetClaim.Spec.StorageClassName
+	volumeSnapshotContentName := snapshot.Spec.Source.VolumeSnapshotContentName
+	if volumeSnapshotContentName == nil {
+		return nil, fmt.Errorf("volumeSnapshotContent not found")
+	}
+	volumeSnapshotContent := &snapshotv1.VolumeSnapshotContent{}
+	exists, err := getResource(ctx, client, snapshot.Namespace, *volumeSnapshotContentName, volumeSnapshotContent)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("volumeSnapshotContent does not exist")
+	}
+	matches := []storagev1.StorageClass{}
+	storageClasses := &storagev1.StorageClassList{}
+	if err := client.List(ctx, storageClasses); err != nil {
+		return nil, err
+	}
+	for _, storageClass := range storageClasses.Items {
+		if storageClass.Provisioner == volumeSnapshotContent.Spec.Driver && storageClass.Name != scName {
+			matches = append(matches, storageClass)
+		}
+	}
+	if len(matches) >= 1 {
+		scName = matches[0].Name
```

Comment:
Maybe sort matches before returning the first one in order to be deterministic.  It's also done in the VolumeSnapshotClass lookup code